gdal/port/cpl_vsil_hdfs.cpp

```diff
@@ -260,8 +267,10 @@ VSIHdfsFilesystemHandler::Open( const char *pszFilename,
 int
 VSIHdfsFilesystemHandler::Stat( const char *pszeFilename, VSIStatBufL *pStatBuf, int)
 {
-  memset(pStatBuf, 0, sizeof(*pStatBuf)); // XXX
+  CPLMutexHolder oHolder( &hMutex );
```

Comment:
poFilesystem might be uninitialized at that point if Open() has never been called